src_c/polygon.c

```diff
@@ -460,9 +460,38 @@ pg_polygon_move_ip(pgPolygonObject *self, PyObject *const *args,
     return RAISE(PyExc_TypeError, "move requires a pair of numbers");
 }
 
+static PyObject *
+pg_polygon_collidepoint(pgPolygonObject *self, PyObject *const *args,
+                        Py_ssize_t nargs)
+{
+    double x, y;
+
+    if (nargs == 1) {
+        if (!pg_TwoDoublesFromObj(args[0], &x, &y)) {
+            goto error;
+        }
+    }
+    else if (nargs == 2) {
+        if (!pg_DoubleFromObj(args[0], &x) || !pg_DoubleFromObj(args[1], &y)) {
+            goto error;
+        }
+    }
+    else {
+        goto error;
+    }
```

Comment:
so far we have used this code a lot of times, i think it is time to put it as a function in `geometry.c`/`geometry.h` and name it like _pgTwoDoublesFromFastcall

Reply:
I think we should do that just not in this PR.

Reply:
I'll open another PR that adds that and applies that where necessary.